### DISCUSSION TITLE
Guide on how to use camelCase application-wide

### DISCUSSION THREAD

**Discussion by D-Marc1** [adonisjs/core#1582] [Help]
What's the best way to use camelCase on the entire Adonis project? Meaning database fields, using the query builder, automatic conversion, serialization, etc.? It seems like Adonis (and Knex.js) is a bit opinionated on casing.  Would be nice to have an easy configuration option to do camelCase for the entire application.

**Comment by popovicieduard** [comment]
You can use this to camelize your requests
```
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

import { camelize } from 'App/Helpers/Formatters'

/**
 * Silent auth middleware can be used as a global middleware to silent check
 * if the user is logged-in or not.
 *
 * The request continues as usual, even when the user is not logged-in.
 */
export default class FormatRequest {
	/**
	 * Handle request
	 */
	public async handle({ request }: HttpContextContract, next: () => Promise<void>) {
		/**
		 * Check if user is logged-in or not. If yes, then `ctx.auth.user` will be
		 * set to the instance of the currently logged in user.
		 */

		request.updateQs(camelize(request.get()))
		request.updateBody(camelize(request.post()))

		await next()
	}
}
```

As for query builders, some allow you to use the model properties and some require to use the db naming convention with underscore (_)

**Comment by D-Marc1** [comment]
@thetutlage Sorry, don't mean to bother you, but was wondering if it would be possible to make an official writeup on how to do camelCase for the entire application. Seems like something many people might want. I saw a Github issue where you showed how to do it for just serialization, but some might need it for the whole project.

**Reply by thetutlage** [reply]
Which version of the framework are you using?

**Reply by D-Marc1** [reply]
Wow, thanks for the quick response! I’m using the preview v5.

**Reply by thetutlage** [reply]
Ohh. Then I believe, it is relatively easy. Can you please share the pointers in which you are struggling? I can then look into writing a blog around the same.

**Reply by janainascal** [reply]
Hi people, I would like that as well.
In my case I use a Repository approach, and I would like that to remove all the database logic from the result I returning, separating layers concerns 😎  . For my current project a simple JSON object with all the model properties , such as return by `toJSON()`, but currently I need to map the snake_case back to camelCase

**Reply by janainascal** [reply]
Also, in the future, I intent to use implement a Clean Architecture with a [DataMapper ](https://martinfowler.com/eaaCatalog/dataMapper.html) between the Domain Model and the DatabaseModel. Basicly the 2 model share an interface (properties only) but theirs logic is isolated.

Interface sharing is not mandatory between layers, but it usually happens

**Comment by janainascal** [comment]
Hi @D-Marc1, I also would like this request, but I dind't find the github issue you mention. Can you share the link?

**Reply by D-Marc1** [reply]
For some reason I can't find the issue, but it shows how to do it with the following in `database.ts`:

```ts
orm: {
  getSerializeAsKey(_, key) {
    return key;
  },
},
```

**Reply by janainascal** [reply]
In that case is the table's columns that uses camelCase, correct?

**Reply by D-Marc1** [reply]
No, this would not make the table column's camelCased. This would simply make the API return back data camelCased.

[Here's an issue](https://github.com/adonisjs/core/issues/699) that discusses how you might do that for table column names, but I felt like I reached a dead end with it, due to some set backs I had with it.

**Reply by janainascal** [reply]
I find a [discussion](https://github.com/adonisjs/core/discussions/2428#discussioncomment-536741) with that code you show, and it is use to map the columns in the database, not the serialization (process of converting a Javascript class instance to a JSON object)

You tested it? It works for serialization as well?

**Reply by D-Marc1** [reply]
The code I provided worked for serialization for me. The only thing it doesn't work for is the pagination

**Reply by thetutlage** [reply]
The release coming on 5th allows configuring the casing for Pagination meta data as well. Docs will cover how to achieve it

**Reply by D-Marc1** [reply]
Love it! Thanks @thetutlage so much for your hard work in continuing to improve this awesome framework!
